protocol-units/bridge/move-modules/sources/atomic_bridge.move

```diff
@@ -0,0 +1,239 @@
+    }
+
+    struct BridgeTransferDetails has key, store {
+        initiator: vector<u8>, // eth address
+        recipient: address,
+        amount: u64,
+        hash_lock: vector<u8>,
+        time_lock: u64,
+    }
+
+    struct BridgeConfig has key {
+        moveth_minter: address,
+        bridge_module_deployer: address,
+    }
+
+    #[event]
+    /// An event triggered upon locking assets for a bridge transfer 
+    struct BridgeTransferAssetsLockedEvent has store, drop {
+        bridge_transfer_id: vector<u8>,
+        caller: address,
+        recipient: address,
+        amount: u64,
+        hash_lock: vector<u8>,
+        time_lock: u64,
+    }
+
+    #[event]
+    /// An event triggered upon completing a bridge transfer
+    struct BridgeTransferCompletedEvent has store, drop {
+        bridge_transfer_id: vector<u8>,
+        secret: vector<u8>,
+    }
+
+    #[event]
+    /// An event triggered upon cancelling a bridge transfer
+    struct BridgeTransferCancelledEvent has store, drop {
+        bridge_transfer_id: vector<u8>,
+    }
+
+    entry fun initialize(deployer: &signer, moveth_minter: address) {
+        let bridge_transfer_store = BridgeTransferStore {
+            pending_transfers: smart_table::new(),
+            completed_transfers: smart_table::new(),
+            aborted_transfers: smart_table::new(),
+        };
+        let bridge_config = BridgeConfig {
+            moveth_minter,
+            bridge_module_deployer: signer::address_of(deployer),
+        };
+        move_to(deployer, bridge_transfer_store);
+        move_to(deployer, bridge_config);
+    }
+
+    public fun lock_bridge_transfer_assets(
+        caller: &signer,
+        initiator: vector<u8>, //eth address
+        bridge_transfer_id: vector<u8>,
+        hash_lock: vector<u8>,
+        time_lock: u64,
+        recipient: address,
+        amount: u64
+    ): bool acquires BridgeTransferStore {
+        let bridge_store = borrow_global_mut<BridgeTransferStore>(signer::address_of(caller));
+        let details = BridgeTransferDetails {
+            recipient,
+            initiator,
+            amount,
+            hash_lock,
+            time_lock: timestamp::now_seconds() + time_lock
+        };
+
+        smart_table::add(&mut bridge_store.pending_transfers, bridge_transfer_id, details);
+        event::emit(
+            BridgeTransferAssetsLockedEvent {
+                bridge_transfer_id,
+                caller: signer::address_of(caller),
+                recipient,
+                amount,
+                hash_lock,
+                time_lock,
+            },
+        );
+
+        true
+    }
+
+    public fun complete_bridge_transfer(
+        caller: &signer,
+        bridge_transfer_id: vector<u8>,
+        secret: vector<u8>
+    ) acquires BridgeTransferStore, BridgeConfig {
+        let bridge_store = borrow_global_mut<BridgeTransferStore>(signer::address_of(caller));
```

Comment:
The client is the one who will call this function. It can be called by anyone who possesses the secret, as you explained. Make sure this is callable by anyone.